src/checkpoint.c

```diff
@@ -72,6 +74,23 @@ static struct argp_option options[]
 
 static char args_doc[] = "checkpoint CONTAINER";
 
+int
+crun_parse_network_lock_method (char *param arg_unused)
+{
+#if HAVE_CRIU && HAVE_DLOPEN
+  if (strcmp (param, "iptables") == 0)
+    return CRIU_NETWORK_LOCK_IPTABLES;
+  else if (strcmp (param, "nftables") == 0)
+    return CRIU_NETWORK_LOCK_NFTABLES;
+  else if (strcmp (param, "skip") == 0)
+    return CRIU_NETWORK_LOCK_SKIP;
```

Comment:
@giuseppe CRIU_NETWORK_LOCK_SKIP was introduced in criu 3.19 and 4.0 and afaict, centos8 is at 3.15, hence the failure.

Reply:
we can solve this with a configure check, we need to check for the presence of `CRIU_NETWORK_LOCK_SKIP` in `configure.ac` and then add this code only when it is available (i.e. `#if HAVE_CRIU && HAVE_DLOPEN & HAVE_CRIU_NETWORK_LOCK_SKIP`)

Reply:
You seem to have used `HAVE_CRIU_NETWORK_LOCK` in the changes instead of `HAVE_CRIU_NETWORK_LOCK_SKIP`. It looks somewhat misleading, intentional?

Reply:
otherwise we would need also to check for LOCK_NFTABLES and LOCK_IPTABLES.  Or could we assume they are always present?
Personally, I don't think it would matter too much, only old systems won't use the new functionality

Reply:
pushed a new version, restricted to NETWORK_LOCK_SKIP.  Let's see if it builds :-)

Reply:
it does.  If you are happy with it, we can merge

Reply:
Looks good, we can merge, thanks!